tests/publiccloud/sles4sap.pm

```diff
@@ -28,48 +28,55 @@ Upload the HA/SAP logs from instance C<$instance> on the Webui.
 =cut
 sub upload_ha_sap_logs {
     my ($self, $instance) = @_;
-    my @logfiles = qw(provisioning.log salt-deployment.log salt-formula.log salt-pre-installation.log);
+    my @logfiles = qw(salt-deployment.log salt-os-setup.log salt-pre-deployment.log salt-result.log);
 
     # Upload logs from public cloud VM
-    $instance->run_ssh_command(cmd => 'sudo chmod o+r /tmp/*.log');
+    $instance->run_ssh_command(cmd => 'sudo chmod o+r /var/log/salt-*');
     foreach my $file (@logfiles) {
-        $instance->upload_log("/tmp/$file", log_name => $instance->{instance_id});
+        $instance->upload_log("/var/log/$file", log_name => $instance->{instance_id});
     }
 }
 
 =head2 wait_until_resources_started
 
-    wait_until_resources_started();
+    wait_until_resources_started( [ timeout => $timeout ] );
 
-Wait for HA cluster to be started.
+Wait for resources to be started. Runs C<crm cluster wait_for_startup> in SUT as well
+as other verifications on newer versions of SLES (12-SP3+), for up to B<$timeout> seconds
+for each command. Timeout must be specified by the named argument B<timeout> (defaults
+to 120 seconds). This timeout is scaled by the factor specified in the B<TIMEOUT_SCALE>
+setting. Croaks on timeout.
 =cut
 sub wait_until_resources_started {
     my ($self, %args) = @_;
     my @cmds    = ('crm cluster wait_for_startup');
     my $timeout = bmwqemu::scale_timeout($args{timeout} // 60);
 
+    # At least one node needs to be online
+    push @cmds, "$crm_mon_cmd | grep -q Node\\ .*\\ online:";
+
     # HANA cluster doesn't restart all resources after fencing in some configurations
     if ($args{cluster_type} ne 'hana') {
         # Some CRM options can only been added on recent versions
         push @cmds, "$crm_mon_cmd | grep -iq no\\ inactive\\ resources" if is_sle '12-sp3+';
+        # This code is a little bit complicated, but we have to invert the RC for the while loop
+        push @cmds, "$crm_mon_cmd | grep -iq starting && RC=false || RC=true; eval \$RC";
     }
 
     # Execute each comnmand to validate that the cluster is running
     # This can takes time, so a loop is a good idea here
     foreach my $cmd (@cmds) {
         # Each command execution has its own timeout, so we need to reset the counter
-        my $starttime = time;
+        my $start_time = time;
 
         # Check for cluster/resources status and exit loop when needed
-        my $ret;
-        while ($ret = $self->{my_instance}->run_ssh_command(cmd => "sudo $cmd", rc_only => 1)) {
+        while ($self->{my_instance}->run_ssh_command(cmd => "sudo $cmd", rc_only => 1)) {
             # Otherwise wait a while if timeout is not reached
-            if (time - $starttime < $timeout) {
+            if (time - $start_time < $timeout) {
                 sleep 5;
             }
             else {
-                # Cluster doesn't start properly, show state before dying
-                $self->run_cmd(cmd => $crm_mon_cmd);
```

Comment:
why you decided to remove this ?

Reply:
First it's because we had the cluster state previously shown, but after reading the code again I think it's better to keep it, thanks!